src/AppMenuSearch.cc

```
@@ -0,0 +1,429 @@
+
+    m_lastSearchQuery = text;
+
+    // Clear results if search text is too short
+    if (isQueryTooShort(text)) {
+        clear();
+        m_lastResults.clear();
+        m_lastProcessedMenu = nullptr;
+        m_lastShowDisabledActions = false;
+        m_lastIgnoreTopLevel = false;
+        m_lastIgnoreSubMenus = false;
+        Q_EMIT repositionRequested();
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)